app/controllers/IndexController.scala

```diff
@@ -16,39 +16,51 @@
 
 package controllers
 
+import config.FrontendAppConfig
 import controllers.actions.IdentifierAction
-import models.UserAnswers
+import models.{Mode, UserAnswers}
 import play.api.Logging
-
-import javax.inject.Inject
 import play.api.i18n.I18nSupport
 import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
 import repositories.SessionRepository
+import services.SubscriptionService
 import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendBaseController
 import views.html.IndexView
 
+import javax.inject.Inject
 import scala.concurrent.{ExecutionContext, Future}
 
 class IndexController @Inject() (
   val controllerComponents: MessagesControllerComponents,
   sessionRepository: SessionRepository,
   identify: IdentifierAction,
+  subscriptionService: SubscriptionService,
+  conf: FrontendAppConfig,
   view: IndexView
 )(implicit ec: ExecutionContext)
     extends FrontendBaseController
     with Logging
     with I18nSupport {
 
-  def onPageLoad: Action[AnyContent] = identify.async {
+  def onPageLoad(mode: Mode): Action[AnyContent] = identify.async {
```

Comment:
I don't think this page needs the concept of a mode. It's always the same content and the same links.

Reply:
Removed it